.github/workflows/package-prepare-release.yml

```diff
@@ -0,0 +1,189 @@
+name: "[Package] Prepare release"
+on:
+  workflow_dispatch:
+    inputs:
+      package:
+        type: choice
+        options:
+        - opentelemetry-propagator-aws-xray
+        - opentelemetry-resource-detector-azure
+        - opentelemetry-sdk-extension-aws
+        description: 'Package to be released'
+        required: true
+
+jobs:
+  prereqs:
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.verify.outputs.version }}
+      changelog: ${{ steps.verify.outputs.changelog }}
+      version_file: ${{ steps.verify.outputs.version_file }}
+      release_branch_name: ${{ steps.verify.outputs.release_branch_name }}
+      next_version: ${{ steps.verify.outputs.next_version }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - id: verify
+        name: Verify prerequisites
+        run: |
+          if [[ $GITHUB_REF_NAME != main ]]; then
+            echo this workflow should only be run against main
+            exit 1
+          fi
+
+          path=./$(./scripts/eachdist.py find-package --package ${{ inputs.package }})
+          changelog=$path/CHANGELOG.md
+
+          if ! grep --quiet "^## Unreleased$" $changelog; then
+            echo the change log is missing an \"Unreleased\" section
+            exit 1
+          fi
+
+          version_dev=$(./scripts/eachdist.py version --package ${{ inputs.package }})
+
+          if [[ ! $version_dev =~ ^([0-9]+)\.([0-9]+)[\.|b]{1}([0-9]+).*.dev$ ]]; then
+            echo "unexpected version: $version"
+            exit 1
+          fi
+
+          version=${version_dev%.dev}
+
+          version_file=$(find $path -type f -name "version.py")
```

Comment:
Let's merge https://github.com/open-telemetry/opentelemetry-python/pull/4099 since it seems to be very close and I'll update this one

Reply:
I just realized that https://github.com/open-telemetry/opentelemetry-python/pull/4099 is in the core repo and there is no PR against contrib.

I updated workflow to support both - `version.py` and `version/__init__py` so it will keep working when contrib is changed to `version/__init__py`. It should be easy to become strict again later one. 

So I'm going to close this one.